.github/workflows/nextflow-tests.yml

```diff
@@ -73,11 +169,257 @@ jobs:
+                  repo: context.repo.repo,
+                  pull_number: ${{ toJSON(needs.discover.outputs.pr_number) }},
+                  commit_id: ${{ toJSON(needs.discover.outputs.review_sha) }},
+                  event: "REQUEST_CHANGES",
+                  body: `*Bleep bloop, I am a robot.*
+
+            ${prefix} the \`${ base_ref }\` branch has test changes that haven't been merged into _this_ branch, and I can't handle that. Please fix that and then I can fix the tests for you!
+
+            \`\`\`console
+            cd /path/to/your/repository/
+            git checkout ${ branch_ref }
+            git fetch origin
+            git merge origin/${ base_ref }
+            git push origin
+            \`\`\``
+                })
+              } else {
+                console.log("Review state not changed")
+              }
+
+              core.setFailed("Need to merge tests")
+            }
+
+      - if: ${{ steps.status.outputs.should-fix == 'true' }}
+        id: pushchanges
+        uses: stefanzweifel/git-auto-commit-action@v5
+        with:
+          commit_message: Autofix Nextflow configuration regression tests
+          branch: ${{ needs.discover.outputs.branch_ref }}
+          file_pattern: '**/configtest*.json'
+
+      - if: ${{ steps.status.outputs.should-fix == 'true' }}
+        name: Add comment about reviewing
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const timers = require('node:timers/promises')
+
+            let commit_id = ${{ toJSON(steps.pushchanges.outputs.commit_hash) }}
+
+            var review_details = {
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: ${{ toJSON(needs.discover.outputs.pr_number) }},
+              commit_id: commit_id,
+              event: "COMMENT",
+              body: `*Bleep bloop, I am a robot.*
+
+            I have updated all of the failing tests for you with ${ commit_id }. You must review my work before merging this pull request!`
+            }
+
+            // We need to trigger GitHub to actually compute the changes
+            // https://docs.github.com/en/rest/guides/using-the-rest-api-to-interact-with-your-git-database
+            for (let i=0; i<10; i++) {
+              // Wait the first time - it will always take some time
+              await timers.setTimeout(3000)
+
+              console.log(`Waited ${i*3} seconds...`)
+
+              var updated_pr = (await github.rest.pulls.get({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                pull_number: ${{ toJSON(needs.discover.outputs.pr_number) }}
+              })).data
+
+              // The SHA must change and the new commit must have settled
+              if ((updated_pr.head.sha != ${{ toJSON(needs.discover.outputs.review_sha) }})
+                  &&
+                  (updated_pr.mergeable != null)) {
+                break
```

Comment:
Should this handle the case of hitting 10 loops and never succeeding?

Reply:
Eh... the only time I've ever seen it take more than 3 seconds is when I get the "no such commit" error above because I made some mistake. I don't know what else to do other than fail if it does take more than 30 seconds.

Reply:
Got it - this means `await github.rest.pulls.createReview(review_details)` will fail anyway? That's fine.

Reply:
Correct - all of the waiting is just trying to ensure that `createReview` won't fail, so in the worst case it does just fail anyway.